addons/udes_stock/models/stock_picking.py

```diff
@@ -850,7 +850,28 @@ def action_confirm(self):
         ):
             pick._create_own_procurement_group()
         res = super().action_confirm()
+        return res
+
+    def _action_done(self):
+        """Extend to collect stats for the UDES monitor"""
+        # Collect action_done stats for monitor
+        with self.statistics() as stats:
+            res = super()._action_done()
 
+        # Ensure when multiple picking types are involved that the names are combined
+        picking_type_names = ""
```

Comment:
Do picking_type_names need a comma to separate them in case there are more than one.
```suggestion
        picking_type_names = ",".join(self.mapped("picking_type_id.name")
```
Remove the for loop in that case

Reply:
Comma works fine, and yes perfect, thank you, I was missing how to do this. Will change soon.

Reply:
Yep happy with changes, just not needed the if because join doesnt add comma when length is 1.
Happy to resolve all threads after that

Reply:
Addressed